### PR TITLE
riscv: Add basic riscv bindings

### PR DESCRIPTION
**Pull request by TimePrinciple**
Updated arm64, x86_64, and added riscv kvm bindings according to v6.9 kernel source.

### Summary of the PR

Generate riscv bindings, and regenerate arm64, x86-64 bindings from v6.9. I've noticed #92, but it has not seen changes for a while, so I decided to push the work. I'm currently working on adding support to `kvm-ioctls` on riscv.

And I wonder how to add cross-compilation test CIs in `rust-vmm-ci`, may be configure a x86_64 ubuntu and set the `runner` field to `/usr/bin/qemu-riscv64` or something?

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by JonathanWoollett-Light** [comment]
We want to integrate testing for RISC-V before merging RISC-V specific code, ideally resolving https://github.com/rust-vmm/rust-vmm-container/issues/96.

**Comment by TimePrinciple** [comment]
> We want to integrate testing for RISC-V before merging RISC-V specific code, ideally resolving [rust-vmm/rust-vmm-container#96](https://github.com/rust-vmm/rust-vmm-container/issues/96).

I've proposed using x86_64 machine to cross-compile and `qemu-riscv64-static` to test compiled binary in [rust-vmm-ci](https://github.com/rust-vmm/rust-vmm-ci/pull/155) and [rust-vmm-container](https://github.com/rust-vmm/rust-vmm-container/pull/101), please have a look at your convenience.

**Comment by TimePrinciple** [comment]
> @TimePrinciple Can you please split the arm64 & x86-64 bindings regenerations into their own separate patches and put them into a new PR as I think it may be some time until the required RISC-V CI enabling is available.

Sure thing. But am I going to do it with the latest kernel version or just stick to v6.9.

**Comment by TimePrinciple** [comment]
It seems that `net.ipv4.ip_forward` is not enabled on runners, which causes the network problem inside QEMU.

I propose two approaches to walk around this:

1. Enable `net.ipv4.ip_forward` by `sysctl -w net.ipv4.ip_forward=1` if it is not enabled.
2. Vendor all dependencies outside QEMU, which could address this problem, but a lot of modifications are to be made.

**Comment by stefano-garzarella** [comment]
> > ~It seems that `net.ipv4.ip_forward` is not enabled on runners, which causes the network problem inside QEMU.~
> > I currently have no clue on what's going on inside the runners, I tried those CI commands locally, everything works fine.
> > I propose vendor all dependencies outside QEMU, which could address this problem, but a lot of modifications are to be made.
> 
> In the container we do `echo 'nameserver 8.8.8.8' > $ROOTFS_DIR/etc/resolv.conf`, I'm not sure if that IP is accessible, so maybe we should use the host as nameserver. In https://wiki.qemu.org/Documentation/Networking it looks like the default virtual DNS server is 10.0.2.3 or `You can specify the guest-visible virtual DNS server address with -netdev user,id=n0,dns=addr`

BTW I'm expecting that the DHCP implemented by the `user` netdev should also set the DNS, so why we need to set `/etc/resolv.conf` ?

**Comment by TimePrinciple** [comment]
I would like to know how do we fix [this](https://buildkite.com/rust-vmm/kvm-bindings-ci/builds/256#0191c6c9-dfdd-42ff-88c6-c304de857710)

**Comment by roypat** [comment]
> I would like to know how do we fix [this](https://buildkite.com/rust-vmm/kvm-bindings-ci/builds/256#0191c6c9-dfdd-42ff-88c6-c304de857710)

You can just adjust the value in the json file, the toolchain update caused this in a few repositories for some reason

**Comment by TimePrinciple** [comment]
It's ready now, please take a look when convenient @stefano-garzarella  @roypat  @rbradford , many thanks 😊

**Review by rbradford — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
+// Allow unused macros and imports for RISC-V since yet implemented
+#![cfg_attr(target_arch = "riscv64", allow(unused_macros, unused_imports))]
```
Comment (roypat) [inline src/lib.rs:11]: Would it make more sense to instead just not have the `serialize` module on riscv64, if we're not implemented serialization support yet? (also, do we ever plan to do serialization support? If so, let's track that in an issue)

Comment (TimePrinciple) [reply src/lib.rs:11]: I'm having some issue when trying to sign the commits while I'm not on my machine but through ssh, I guess I can only fix this tomorrow when I head back 😥

**Review by roypat — APPROVED** [review]

**Review by rbradford — APPROVED** [review]

**Pull request merged**